src/CompiledSouffle.h

```diff
@@ -234,4 +234,97 @@ class t_nullaries {
     void printHintStatistics(std::ostream& o, std::string prefix) const {}
 };
 
+/** info relations */
+template <int Arity>
+class t_info {
+private:
+    std::vector<ram::Tuple<RamDomain, Arity>> data;
+    Lock insert_lock;
+
+public:
+    t_info() = default;
+    using t_tuple = ram::Tuple<RamDomain, Arity>;
+    struct context {};
+    context createContext() {
+        return context();
+    }
+    class iterator : public std::iterator<std::forward_iterator_tag, ram::Tuple<RamDomain, Arity>> {
+        typename std::vector<ram::Tuple<RamDomain, Arity>>::const_iterator it;
+
+    public:
+        iterator(const t_tuple* o = nullptr) : it(o) {}
+
+        const t_tuple operator*() {
+            return *it;
+        }
+
+        bool operator==(const iterator& other) const {
+            return other.it == it || *other.it == *it;
+        }
+
+        bool operator!=(const iterator& other) const {
+            return !(*this == other);
+        }
+
+        iterator& operator++() {
+            it++;
+            return *this;
+        }
+    };
+    iterator begin() const {
+        return iterator(data.data());
+    }
+    iterator end() const {
+        return iterator(data.data() + data.size());
+    }
+    void insert(const t_tuple& t) {
+        insert_lock.lock();
```

Comment:
Please check whether it is contained in the vector first before inserting it into the vector.